### ISSUE TITLE
[BUG] Add to Calendar toggle when off does not work

### ISSUE THREAD

**Issue by vburgett** [rokwire/illinois-app#3789]
Under Settings > My Calendar Settings, "Prompt when saving events or appointments to calendar" is toggled off and I can not toggle it on with a tap. On an event details panel, the Add to Calendar option does nothing when tapped. 

v 5.0.74
iphone 13 Pro 16.6.1

Can you match the below? Please replace "My Calendar Settings" with "Add to My Device's Calendar" 
Zeplin Events 5.0 files provided 5/2023: https://zpl.io/YYjWJDd

<img width="1567" alt="Screenshot 2023-10-05 at 10 52 59 AM" src="https://github.com/rokwire/illinois-app/assets/1044905/fe906b3c-2998-458c-917f-191809bb52ea">

**Comment by TodorBachvarov** [comment]
Hi @vburgett, regarding ""Prompt when saving events or appointments to calendar" is toggled off and I can not toggle it on with a tap" - I'm not able to reproduce it, for both Dev and Prod the switch changes his state when tapped (also correctly show/don't show the prompt dialog depending on the state of the switch). 
I've found and fix small issue that was preventing the app from adding event to the device calendar. 
I've updated the settings title  "My Calendar Settings" to "Add to My Device's Calendar". 
Please retest with version 5.0.77

**Comment by vburgett** [comment]
I am still having this issue in 5.0.78. The "off" toggle is set to "off" for the setting "prompt when saving events or appointments to calendar" and does nothing when tapped. 

The "Add to Calendar" option on an event also does nothing when tapped.

On 10/14 during the IlliniFest event, the event coordinators shared that attendees reported "Add to Calendar didn't work on iPhones." I asked three other Rokwire colleagues to test this and they did not have this issue. At least one colleague was on an iPhone iOS 16.6.1.
 
Illinois app v5.0.78
iPhone 13 Pro
iOS 16.6.1

**Comment by mihail-varbanov** [comment]
@vburgett, I was not able to recreate the first issue that you talk about. Tapping on "Prompt when saving events or appointments to calendar" toggles the selection on/off for me.

Regarding the second one "The "Add to Calendar" option on an event also does nothing when tapped.", the problem was that we did not report the result of the operation, that left the user with impression that it does not work. I added a message prompt that reports both positive and negative results. This is available in version 5.1.8.

On the third issue that you talk about "Add to Calendar didn't work on iPhones.", most probably it is a duplication of the above issue.

**Comment by vburgett** [comment]
Thank you, @mihail-varbanov. Since this is an issue on the iPhone, do I need to wait for the 5.1 release to test it?

Can the "Add to My Device's Calendar" settings match the 5.0 design? Otherwise, where is the user giving or retracting permission for the app to have access to their calendar? 

My iPhone's privacy settings for my calendar and the Illinois app were set to "None." I changed the setting to "Full Access" and the Illinois app still is not responsive for the Add to Calendar or the calendar setting toggle. You can see in the screenshare below that when I tap on those links, they do not even show up as taps. Both links are not responsive. I checked and no events were added to my device's calendar.

v5.078
iphone 13 pro
iOS 17.2.1

![Add to Calendar](https://github.com/rokwire/illinois-app/assets/1044905/21c4f737-956d-441e-bcd2-ce564bcc93ff)

https://github.com/rokwire/illinois-app/assets/1044905/8454f0fb-7b13-4bb8-8948-e81e16336de4

**Comment by mihail-varbanov** [comment]
Hi @vburgett,

I had to cleaned up our calendar service in the Illinois app because it was a big mess. I think now everything works as it should do. This is available in version 5.1.10.

Regarding to testing the app on iOS, if you send me your iPhone's UDID I can add it to the registered devices in the University development portal, so that you will be able to install the developers builds that we create regularly. You can find your iPhone's UDID in Xcode, iTunes or Finder as shown in the screenshot bellow, or you can load  https://get.udid.io/ in your iPhones browser and follow the instructions.

<img width="1001" alt="Screenshot 2024-01-18 at 5 36 07 PM" src="https://github.com/rokwire/illinois-app/assets/5631487/2f59c7a4-99b6-4444-8502-4bbb65df57eb">

**Comment by vburgett** [comment]
00008110-000E613C3E22801E

**Comment by mihail-varbanov** [comment]
@vburgett, I registered your iPhone 13 Pro in the University devices list. Now you should be able to install our dev builds on this device, starting from version 5.1.11. @pmarkhennessy can share with you the link to the today's iOS daily build that we create by the end of the day. The easiest way to install the iOS app is to scan the QR code that we create.

**Comment by vburgett** [comment]
- I was able to install 5.1.13 on my iPhone. 
- I _could_ toggle on and off the "Add to My Device's Calendar" "Prompt when saving events . . ."
- I could _not_ save an event to my calendar. I received the below error. Please correct the misspelling in the error message to read "Permission denied to add the event to your calendar."  
- When I looked in my phone's settings to see the access the Illinois app had to my calendar, Illinois wasn't even listed as an option. See screenshot. Do we need an additional setting under ""Add to My Device's Calendar" that says "Allow access to my calendar"?

The events I tried to add to my calendar were: 
- Test DD registration upload (1/25-1/26)
- Gather in the Atrium (1/30)
- Cookie Collab (2/15)
- Money in Your Entrepreneurial Pursuit (1/24)

![IMG_2784](https://github.com/rokwire/illinois-app/assets/1044905/6cfc5039-7989-4cf7-a850-7bbbf8a426b3)
![IMG_2785](https://github.com/rokwire/illinois-app/assets/1044905/d7520e7b-0e93-40b9-82a7-260f30046bd3)
![IMG_2786](https://github.com/rokwire/illinois-app/assets/1044905/9a1eba73-70a5-468c-8658-04ccba0072e3)
![IMG_2787](https://github.com/rokwire/illinois-app/assets/1044905/8678bf95-afa6-4444-87d8-8f037506daaa)
@mihail-varbanov

**Comment by mihail-varbanov** [comment]
Hi @vburgett, I checked this again and I did not see any error:

**1. Right after a fresh install, if you check the Illinois app permissions in the iOS Settings app, the"Calendar" permission option will not be listed as we never tried to access the Calendar service yet:**

![Screen Shot 2024-01-28 at 11 09 32 AM](https://github.com/rokwire/illinois-app/assets/5631487/5f80109c-3082-49a0-b44b-23a10a5e9a09)


**2. When you tap on "Add to Calendar" for the first time, we request permission from the operating system to access the Calendar service. The operating system, in turn, prompts the user to grant this permission:**

![Screen Shot 2024-01-28 at 10 54 55 AM](https://github.com/rokwire/illinois-app/assets/5631487/a56919e5-09f7-45fc-abbf-bc75369eb92f)


**3. If you grant the permission the operation succeeds:**

![Screen Shot 2024-01-28 at 11 04 00 AM](https://github.com/rokwire/illinois-app/assets/5631487/948f2bd2-9f36-43d3-813d-6fedf898f43f)

and the app's settings page looks like this:

![Screen Shot 2024-01-28 at 10 58 18 AM](https://github.com/rokwire/illinois-app/assets/5631487/eb0a3308-1115-4460-9258-7318550a448b)


**4. If you don't grant this permission, it fails:**

![Screen Shot 2024-01-28 at 11 06 30 AM](https://github.com/rokwire/illinois-app/assets/5631487/6d9ec032-c613-472d-851e-7bfac8c72372)

and the app's settings page looks like this:

![Screen Shot 2024-01-28 at 11 23 47 AM](https://github.com/rokwire/illinois-app/assets/5631487/2a19cda2-367b-4707-abc2-8fdea912aed3)


**-------------------------------------**

In your case, it seems the Illinois app was not granted permission to access Calendar service. I am not sure where the "Calendars" snapshot had been taken. The most reliable source to get iOS permission settings is the app's page in the iOS Settings app.

**Comment by vburgett** [comment]
@mihail-varbanov I downloaded 5.1.16 for iOS today. When tapping "Add to Calendar" for the first time after the install, I did _not_ get the screen you show above that says "Illinois would like to access your calendar." I continue to get the denied screen with no event added to my device's calendar.

There is a misspelling in the error message. Please update it to read "Permission denied to add the event to your calendar."

**Comment by mihail-varbanov** [comment]
@vburgett,

1. What iOS version does your iPhone have?
2. Did you uninstall the Illinois app before installing 5.1.16?
3. Can you please me a snapshot of Illinois app settings page in the iOS Settings app (the last screen from my previous post)?

**Comment by vburgett** [comment]
1. iOS 17.2.1 (iPhone 13 Pro)

2. Yes
Here are the steps I take:
Uninstall Illinois 5.0.78 by deleting the app and, according to iOS, all my data
Change Settings > Privacy & Security > Developer Mode to "on"
Restart my iPhone.
Install dev build.

3. See below screenshot.

I tried this process again with the same error and misspelling to correct. I used 5.1.18.

![IMG_2930](https://github.com/rokwire/illinois-app/assets/1044905/9316701c-ee9b-4f7f-9cc1-8255049b1b17)
